### ISSUE TITLE
WorkspaceFileGenerator uses wrong encoding for StringInputStream

### ISSUE THREAD

**Issue by hannesN** [eclipse/xtext-eclipse#1566]
We use a NewFileWizard which is supposed to create a file containing an UTF-8 String.
However, the FileGenerator is creating a StringInputStream with the default charset, on windows "windows-1252" therefore creating invalid input.

I suggest using the charset specified in the IFile to create. 
e.g.
```
is = new StringInputStream(fileEntry.getValue().toString(), file.getCharset());
file.create(is, true, subMonitor);
``` 

I tested it with my code and it works for me.

**Comment by cdietrich** [comment]
can you please provide a PR

**Comment by cdietrich** [comment]
cc @ArneDeutsch

**Comment by hannesN** [comment]
I created a PR.

**Comment by cdietrich** [comment]
fixed in 2.24
